src/localsocketcontroller.cpp

```diff
@@ -105,6 +108,7 @@ void LocalSocketController::initializeInternal() {
 #endif
 
   logger.debug() << "Connecting to:" << path;
+  m_socket->disconnectFromServer();
```

Comment:
Can this hang or wait? The docs indicate that this can wait. Do we know if it can handle a crashed daemon if the socket has data waiting to be read?

Reply:
When the daemon crashes, writing to the socket would expect to throw a `SIGPIPE` because there is no longer anything on the reading end of the pipe, so I think that it cannot hang in the event of a crash. If the daemon becomes unresponsive then this might result in a hang, so I guess `QLocalSocket::abort()` is probably the better function to call here.